### PR TITLE
tc: Add test for RemoveChild()

### PR DESCRIPTION
**Pull request by changlan**
This PR adds tests for `RemoveChild()` in `traffic_class`, as a part of the ongoing effort to improve code coverage.

### CODE REVIEW

File: core/traffic_class_test.cc [hunk 1]
```diff
@@ -22,8 +22,8 @@ class DummyModule : public Module {
   struct task_result RunTask(void *arg) override;
 };
 
-[[gnu::noinline]] struct task_result DummyModule::RunTask(
-    [[maybe_unused]] void *arg) {
+[[gnu::noinline]] struct task_result DummyModule::RunTask([
```
Comment (barath) [inline core/traffic_class_test.cc]: Is clang format doing something strange here?

**Comment by barath** [comment]
LGTM

File: core/traffic_class_test.cc (hunk 1 above)
Comment (changlan) [inline core/traffic_class_test.cc]: Yes, it is formatted by `clang-format`. I feel that's strange too.

**Comment by barath** [comment]
photo 197

**Pull request merged**